gym/wrappers/frame_skipping.py

```diff
@@ -0,0 +1,34 @@
+import gym
+
+__all__ = ['SkipWrapper']
+
+def SkipWrapper(repeat_count):
+    class SkipWrapper(gym.Wrapper):
+        """
+            Generic common frame skipping wrapper
+            Will perform action for `x` additional steps
+        """
+        def __init__(self, env):
+            super(SkipWrapper, self).__init__(env)
+            self.repeat_count = repeat_count
+            self.stepcount = 0
+
+        def _step(self, action):
+            done = False
+            total_reward = 0
+            current_step = 0
+            while current_step < (self.repeat_count + 1) and not done:
+                self.stepcount += 1
+                obs, reward, done, info = self.env.step(action)
+                if 'skip.stepcount' not in info:
```

Comment:
Looks like you made the change. Will merge when back at a computer. Thanks!

On Sunday, August 14, 2016, Greg Brockman <gdb@openai.com> wrote:

> I guess my point is, when would it be desirable? In general, anything we
> make possible today we're going to have to support forever, so we should be
> conservative.
>
> (Normally I wouldn't even want to merge a given wrapper until I've seen a
> few cases, but given the wrapper feature in its entirety is experimental, I
> think we have a bit more flexibility for now.)
>
> On Sunday, August 14, 2016, Philip Paquette <notifications@github.com
> <javascript:_e(%7B%7D,'cvml','notifications@github.com');>> wrote:
>
>> In gym/wrappers/frame_skipping.py
>> <https://github.com/openai/gym/pull/293#discussion_r74710570>:
>>
>> > +            Generic common frame skipping wrapper
>> > +            Will perform action for `x` additional steps
>> > +        """
>> > +        def __init__(self, env):
>> > +            super(SkipWrapper, self).__init__(env)
>> > +            self.repeat_count = repeat_count
>> > +            self.stepcount = 0
>> > +
>> > +        def _step(self, action):
>> > +            done = False
>> > +            total_reward = 0
>> > +            current_step = 0
>> > +            while current_step < (self.repeat_count + 1) and not done:
>> > +                self.stepcount += 1
>> > +                obs, reward, done, info = self.env.step(action)
>> > +                if 'skip.stepcount' not in info:
>>
>> The only way they can be nested is if someone includes one by default in
>> an env metadata.
>>
>> —
>> You are receiving this because you commented.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/openai/gym/pull/293/files/34eae26273b873e28ad4e2db18e94e446526c9ff#r74710570>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/AAM7kYSUGdFtb8toHxTbPLFAu_0ZvlaLks5qf6ENgaJpZM4Jj_1O>
>> .
>>
>
>
> --
> Sent from mobile
>


-- 
Sent from mobile